extensions/utm-tracker/src/index.ts

```diff
@@ -0,0 +1,59 @@
+import { register } from '@shopify/web-pixels-extension';
+
+type utmTracker = {
+   utm_source?: string;
+   utm_medium?: string;
+   utm_campaign?: string;
+   utm_content?: string;
+   timestamp: number;
+};
+register(({ analytics, browser, init, settings }): void => {
+   const UTM_TRACKERS_KEY = 'utmTrackers';
+
+   const storeUtmTrackers = (): void => {
+      const url = new URL(init.context.document.location.href);
+      const utmTrackers = url.searchParams.get(UTM_TRACKERS_KEY);
+      if (utmTrackers) {
+         void browser.localStorage
+            .setItem(UTM_TRACKERS_KEY, utmTrackers)
+            .catch((error) => console.error(error));
+      } else {
+         void browser.localStorage
+            .removeItem(UTM_TRACKERS_KEY)
+            .catch((error) => console.error(error));
+      }
+   };
+
+   const sendOrderAttributions = async (orderid?: string): Promise<void> => {
+      const utmTrackers = (await browser.localStorage.getItem(
+         UTM_TRACKERS_KEY
+      )) as string | null;
+      if (utmTrackers && orderid) {
+         const jsonUtmTrackers = JSON.parse(utmTrackers) as utmTracker[];
+         await fetch(
+            `${settings.ifixit_base_url}/api/2.0/cart/order/attributions/create`,
+            {
+               method: 'POST',
+               body: JSON.stringify({
+                  shopifyOrderId: orderid,
+                  [UTM_TRACKERS_KEY]: jsonUtmTrackers,
+               }),
+            }
+         );
+      }
+   };
+
+   analytics.subscribe('checkout_started', (): void => {
+      try {
+         storeUtmTrackers();
+      } catch (e) {
+         console.error(e);
+      }
+   });
+
+   analytics.subscribe('checkout_completed', (event): void => {
+      void sendOrderAttributions(event.data.checkout.order?.id).catch((error) =>
+         console.error(error)
+      );
+   });
```

Comment:
This is an alternative way to circumvent the eslint error, since it doesn't seem that `analytics.subscribe()` accepts async functions. They are floating promises but it should be fine.

```suggestion
   analytics.subscribe('checkout_started', () => {
      storeUtmTrackers().catch((error) => console.error(error));
   });

   analytics.subscribe('checkout_completed', (event) => {
      sendOrderAttributions(event.data.checkout.order?.id).catch((error) =>
         console.error(error)
      );
   });
```